google/cloud/bigquery/_job_helpers.py

```diff
@@ -197,14 +198,9 @@ def _to_query_job(
     job_complete = query_response.get("jobComplete")
     if job_complete:
         query_job._properties["status"]["state"] = "DONE"
-        # TODO: https://github.com/googleapis/python-bigquery/issues/589
-        # Set the first page of results if job is "complete" and there is
-        # only 1 page of results. Otherwise, use the existing logic that
-        # refreshes the job stats.
-        #
-        # This also requires updates to `to_dataframe` and the DB API connector
-        # so that they don't try to read from a destination table if all the
-        # results are present.
+        query_job._query_results = google.cloud.bigquery.query._QueryResults(
```

Comment:
Does this change mean we also load the query results when job is complete, when `query(api_method="INSERT")`?

Reply:
Good question. I just double-checked that this method is only called from `query_jobs_query`, so it won't affect when `api_method="INSERT"`